plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/gitlabMergeRequest.ts

```diff
@@ -148,7 +156,7 @@ export const createPublishGitlabMergeRequestAction = (options: {
       });
 
       const actions: Types.CommitAction[] = fileContents.map(file => ({
-        action: 'create',
+        action: ctx.input.commitAction ?? 'create',
```

Comment:
Just to mention, I have tested Gitlab API, and if the action doesn't match the state of the file (i.e. if it already exists or not), then the API call will fail with HTTP 400, and that will cause the action to fail

Reply:
> or we could somehow ask the user to provide a map of actions as an input but I think that would be way too complicated and not that user-friendly.

Yeah I agree this would not be user-friendly. Sucks! I guess one way would be to check if the file already exists and then give it the right action type. But that wouldn't be very scalable and could slow it down 🤔

Reply:
Huge disclaimer that I don't know too much about the gitlab workings but its looks very strange from an API surface to have an option named commitAction taking create/update/delete, I don't really understand what it's doing from looking at this action 😅 If delete is unsupported I don't think we should expose this from the start.

Reply:
Thanks for your feedback, I agree that it's not a great interface. Do you have any suggestions? The only way to completely remove that option is to implement something similar to what @djamaile suggested above but I think that would make a breaking change as up until now users have been using this action to add new files to existing repositories and if we add a feature where we are detecting if something was changed/deleted that might create merge requests that contain undesired changes i.e. most likely removals of existing files.

Reply:
@ljiljanaholden `Delete` is supported by the gitlab API, see `action` in [docs](https://docs.gitlab.com/ee/api/commits.html). Here you can see create, delete, move, update, chmod are all supported but take different inputs as per their example.